File: packages/hmssdk_flutter/ios/Classes/Models/polls/HMSPollAnswerResponseExtension.swift
//
//  HMSPollAnswerResponseExtension.swift
//  hmssdk_flutter
//
//  Created by Pushpam on 31/01/24.
//

import Foundation
import HMSSDK

class HMSPollAnswerResponseExtension {

    static func toDictionary(pollAnswerResponse: HMSPollQuestionResponseResult) -> [String: Any?] {

        var map = [String: Any?]()

        map["question_index"] = pollAnswerResponse.question
        if let error = pollAnswerResponse.error {
            map["error"] = HMSErrorExtension.toDictionary(error)
        }
        map["correct"] = pollAnswerResponse.correct

        return map

    }

}

File: packages/hmssdk_flutter/android/src/main/kotlin/live/hms/hmssdk_flutter/poll_extension/HMSPollAnswerResponseExtension.kt
package live.hms.hmssdk_flutter.poll_extension

import live.hms.hmssdk_flutter.HMSExceptionExtension
import live.hms.video.polls.models.answer.PollAnswerResponse

class HMSPollAnswerResponseExtension {
    companion object {
        fun toDictionary(pollAnswerResponse: PollAnswerResponse): HashMap<String, Any?>? {
            val map = HashMap<String, Any?>()
            val resultMapList = ArrayList<HashMap<String, Any?>>()
            pollAnswerResponse.result.forEach {
                val resultMap = HashMap<String, Any?>()
                resultMap["correct"] = it.correct
                resultMap["error"] = HMSExceptionExtension.toDictionary(it.error)
                resultMap["question_index"] = it.questionIndex

                resultMapList.add(resultMap)
            }

            map["result"] = resultMapList

            return map
        }
    }
}
